pages/api/stripe/donations/delete.js

```
@@ -3,25 +3,69 @@ import CookieService from '../../../../lib/cookie'
 
 const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY)
 
-const deleteDonation = async ({ email }) => {
-  const customerRes = await stripe.customers.list({ email })
-  const customer = customerRes.data[0]
+const deleteDonation = async ({ email, subscriptionId }) => {
+  try {
+    if (!subscriptionId) {
+      const customers = await stripe.customers.list({
+        email
+      })
+      if (!customers.data?.length) {
+        throw new Error('No customer found with this email')
+      }
 
-  const currentSubscription = customer.subscriptions.data[0]
+      const subscriptions = await stripe.subscriptions.list({
+        customer: customers.data[0].id,
+        status: 'active',
+        limit: 1
+      })
+      if (!subscriptions.data.length) {
+        throw new Error('No active subscription found for this customer')
+      }
 
-  if (!currentSubscription) return
+      subscriptionId = subscriptions.data[0].id
+    }
 
-  // Delete the customers subscription
-  await stripe.subscriptions.del(currentSubscription.id)
+    await stripe.subscriptions.cancel(subscriptionId, {
+      invoice_now: true,
+      prorate: false
+    })
+  } catch (error) {
+    throw new Error('Failed to delete donation')
```

here you are catching the errors you through above, but swallowing the custom messages you made on line 22 and 13.

on line 33 i think you should do 
```
throw new Error(`Failed to delete donation: ${error.message}`)
```

---

you'll want to update the other try catch blocks you have as well to make sure you don't swallow your custom error handling

---

> here you are catching the errors you through above, but swallowing the custom messages you made on line 22 and 13.
> 
> on line 33 i think you should do
> 
> ```
> throw new Error(`Failed to delete donation: ${error.message}`)
> ```

Hehehe... This is an interesting oversight on my part. Let me correct that.